### PR TITLE
feat: add timezone configuration for correct date boundaries

### PR DESCRIPTION
**Pull request by askb**
## Summary

Adds a `user_timezone` configuration option to fix date boundary drift for non-UTC users (e.g., Brisbane UTC+10 has 14 hours where UTC 'today' ≠ local 'today').

### Config
- New `user_timezone` option in addon settings (default: `UTC`)
- Supports any IANA timezone (e.g., `Australia/Brisbane`, `America/New_York`)

### garmin-sync.py
- `_user_today()` helper for timezone-aware date boundaries
- `_extract_sleep_time()` now uses explicit `utcfromtimestamp()` for Garmin Local timestamps (documents intent, avoids system-TZ dependency)
- All sync date calculations use user timezone

### metrics-compute.py
- Reads `USER_TIMEZONE` env var for consistent date handling
- Logs timezone on startup

### ha-notify.py
- All 7 HA sensors now include `timezone` and `last_computed` attributes
- Reads `USER_TIMEZONE` env var

### s6 service
- Exports `USER_TIMEZONE` from addon config to all Python services

### Tests
- 5 new timezone tests (sleep time extraction, date boundaries, default UTC)
- Fixed 3 sleep time accuracy tests to use UTC-aware epoch construction

## Version
0.11.4 → 0.12.0 (new feature)

Refs: #78

### CODE REVIEW

File: tests/test_garmin_sync.py [hunk 1]
```diff
@@ -563,3 +563,50 @@ def test_handles_db_error_gracefully(
...
+    def test_user_today_respects_timezone(self, garmin_sync):
+        """_user_today returns date in the configured timezone."""
+        from datetime import date
+        today = garmin_sync._user_today()
+        assert isinstance(today, date)
+
+    def test_user_tz_defaults_to_utc(self, garmin_sync):
+        """USER_TZ defaults to UTC when no environment variable is set."""
+        from zoneinfo import ZoneInfo
+        assert garmin_sync.USER_TZ == ZoneInfo("UTC")
```
[the remark this answers is not in the document]
Comment (askb) [reply tests/test_garmin_sync.py]: Fixed — now uses str() comparison instead of identity. Also improved the _user_today test to validate timezone boundary behavior with Pacific/Auckland patch.

File: pulsecoach/rootfs/app/scripts/ha-notify.py [hunk 2]
```diff
@@ -59,6 +68,9 @@ def ha_request(method: str, path: str, data: dict | None = None) -> dict | None:
 
 def push_sensor(entity_id: str, state: str | float | int, attributes: dict) -> bool:
     """Push a sensor state to HA."""
+    # Include timezone and last_updated in all sensor attributes
```
[the remark this answers is not in the document]
Comment (askb) [reply pulsecoach/rootfs/app/scripts/ha-notify.py]: The comment already says last_computed which matches the attribute key. No change needed.

File: pulsecoach/rootfs/app/scripts/garmin-sync.py [hunk 3]
```diff
@@ -167,13 +181,17 @@ def _safe_sleep_minutes(sleep_dto, key):
 
 
 def _extract_sleep_time(sleep_dto, key):
-    """Extract sleep timestamp and convert to minutes-from-midnight string."""
+    """Extract sleep timestamp and convert to minutes-from-midnight string.
+
+    Garmin's *Local timestamps are epoch-ms adjusted to the user's local time,
+    so utcfromtimestamp gives the local wall-clock hours/minutes directly.
```
[the remark this answers is not in the document]
Comment (askb) [reply pulsecoach/rootfs/app/scripts/garmin-sync.py]: Fixed — expanded docstring to explain Garmin's encoding: 22:30 AEST stored as epoch-ms for 22:30 UTC, and why utcfromtimestamp recovers the correct wall-clock time.

File: tests/test_garmin_sync.py [hunk 4]
```diff
@@ -563,3 +563,50 @@ def test_handles_db_error_gracefully(
...
+        """_extract_sleep_time returns None for missing timestamps."""
+        assert garmin_sync._extract_sleep_time({}, "sleepStartTimestampLocal") is None
+        assert garmin_sync._extract_sleep_time({"sleepStartTimestampLocal": None}, "sleepStartTimestampLocal") is None
+
+    def test_user_today_respects_timezone(self, garmin_sync):
+        """_user_today returns date in the configured timezone."""
+        from datetime import date
+        today = garmin_sync._user_today()
+        assert isinstance(today, date)
+
```
[the remark this answers is not in the document]
Comment (askb) [reply tests/test_garmin_sync.py:622]: Fixed — test now validates UTC date match and patches USER_TZ to Pacific/Auckland to verify timezone boundary behavior.
